### ISSUE TITLE
Parsing error on generic classes in .svelte.ts

### ISSUE THREAD

**Issue by WarningImHack3r** [sveltejs/eslint-plugin-svelte#1079]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.21.0

### What version of `eslint-plugin-svelte` are you using?

3.0.0-next.18

### What did you do?

<details>
<summary>Configuration</summary>

```
bun lint in a SvelteKit project
```
</details>

```ts
// local-store.svelte.ts

export class LocalStore<T> { // parsing error on `<`: unexpected token
	value = $state<T>() as T;
	key = "";

	constructor(key: string, value: T) {
		this.key = key;
		this.value = value;
	}
}
```

### What did you expect to happen?

The file to get parsed correctly

### What actually happened?

A parse error when running the command line and in the IDE

### Link to **GitHub Repo** with Minimal Reproducible Example

The snippet I provided is all you should need, create such a file with the right name (only breaks when using `.svelte.ts`, not simply `.ts`, and you can instantly see your IDE complaining and the command line reporting an issue
Tell me if you really need a full-fledged repo for that still

**Comment by baseballyama** [comment]
Please provide eslint config.

**Comment by WarningImHack3r** [comment]
Sorry, here it is (basically the default SveltKit config):
```js
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import svelte from "eslint-plugin-svelte";
import prettier from "eslint-config-prettier";
import globals from "globals";
import { fileURLToPath } from "node:url";
import { includeIgnoreFile } from "@eslint/compat";

const gitignorePath = fileURLToPath(new URL("./.gitignore", import.meta.url));

export default tseslint.config(
        includeIgnoreFile(gitignorePath),
        eslint.configs.recommended,
        tseslint.configs.recommended,
        svelte.configs["flat/recommended"],
        prettier,
        svelte.configs["flat/prettier"],
        {
                languageOptions: {
                        globals: {
                                ...globals.browser,
                                ...globals.node
                        }
                }
        },
        {
                files: ["**/*.svelte"],
                languageOptions: {
                        parserOptions: {
                                parser: tseslint.parser
                        }
                }
        }
);
```

**Comment by baseballyama** [comment]
Please use `typescript-eslint`. For more details please read README.

**Comment by baseballyama** [comment]
But I think we need to use `svelte-eslint-parser` for parser.
Can you read README once again?

**Comment by WarningImHack3r** [comment]
I'll try but I mean... It's the configuration SvelteKit comes with. How could they mess that up?

**Comment by baseballyama** [comment]
I will update kit repo when `eslint-plugin-svelte` v3 is released.👍

**Comment by WarningImHack3r** [comment]
Update: I've tried it and it doesn't seem to work either... it even fails to parse my `+page.svelte` file now. I followed [this](https://github.com/sveltejs/eslint-plugin-svelte#parser-configuration).

```javascript
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import svelte from "eslint-plugin-svelte";
import svelteParser from "svelte-eslint-parser";
import prettier from "eslint-config-prettier";
import globals from "globals";
import { fileURLToPath } from "node:url";
import { includeIgnoreFile } from "@eslint/compat";

const gitignorePath = fileURLToPath(new URL("./.gitignore", import.meta.url));

export default tseslint.config(
	includeIgnoreFile(gitignorePath),
	eslint.configs.recommended,
	tseslint.configs.recommended,
	svelte.configs.recommended,
	prettier,
	svelte.configs.prettier,
	{
		languageOptions: {
			globals: {
				...globals.browser,
				...globals.node
			}
		}
	},
	{
		files: ["**/*.svelte"],
		languageOptions: {
			parser: svelteParser,
			parserOptions: {
				parser: tseslint.parser,
				project: "./tsconfig.json",
				extraFileExtensions: [".svelte"]
			}
		}
	}
);
```
Diff:
```diff
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import svelte from "eslint-plugin-svelte";
+import svelteParser from "svelte-eslint-parser";
import prettier from "eslint-config-prettier";
import globals from "globals";
import { fileURLToPath } from "node:url";
import { includeIgnoreFile } from "@eslint/compat";

const gitignorePath = fileURLToPath(new URL("./.gitignore", import.meta.url));

export default tseslint.config(
	includeIgnoreFile(gitignorePath),
	eslint.configs.recommended,
	tseslint.configs.recommended,
-	svelte.configs["flat/recommended"],
+	svelte.configs.recommended,
	prettier,
-	svelte.configs["flat/prettier"],
+	svelte.configs.prettier,
	{
		languageOptions: {
			globals: {
				...globals.browser,
				...globals.node
			}
		}
	},
	{
		files: ["**/*.svelte"],
		languageOptions: {
+			parser: svelteParser,
			parserOptions: {
				parser: tseslint.parser,
+				project: "./tsconfig.json",
+				extraFileExtensions: [".svelte"]
			}
		}
	}
);
```

**Comment by baseballyama** [comment]
Please read https://github.com/sveltejs/eslint-plugin-svelte?tab=readme-ov-file#specify-svelteconfigjs also.
We need to add `'**/*.svelte.ts', '*.svelte.ts', '**/*.svelte.js', '*.svelte.js'`.

**Comment by WarningImHack3r** [comment]
This config works:
```js
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import svelte from "eslint-plugin-svelte";
import prettier from "eslint-config-prettier";
import globals from "globals";
import { fileURLToPath } from "node:url";
import { includeIgnoreFile } from "@eslint/compat";
import svelteConfig from "./svelte.config.js";

const gitignorePath = fileURLToPath(new URL("./.gitignore", import.meta.url));

export default tseslint.config(
	includeIgnoreFile(gitignorePath),
	eslint.configs.recommended,
	tseslint.configs.recommended,
	svelte.configs.recommended,
	prettier,
	svelte.configs.prettier,
	{
		languageOptions: {
			globals: {
				...globals.browser,
				...globals.node
			}
		}
	},
	{
		files: ["**/*.svelte", "**/*.svelte.js", "**/*.svelte.ts"],
		languageOptions: {
			parserOptions: {
				parser: tseslint.parser,
				svelteConfig
			}
		}
	}
);
```

However, adding `svelte-eslint-parser`
```diff
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import svelte from "eslint-plugin-svelte";
+import svelteParser from "svelte-eslint-parser";
import prettier from "eslint-config-prettier";
import globals from "globals";
import { fileURLToPath } from "node:url";
import { includeIgnoreFile } from "@eslint/compat";
import svelteConfig from "./svelte.config.js";


export default tseslint.config(
	...
	{
		files: ["**/*.svelte", "**/*.svelte.js", "**/*.svelte.ts"],
		languageOptions: {
+			parser: svelteParser,
			parserOptions: {
				parser: tseslint.parser,
				svelteConfig
			}
		}
	}
);
```
throws the following error running `bun lint`:
```
/path/to/project/src/routes/+page.svelte
  0:0  error  Parsing error: Cannot read properties of null (reading 'end')

✖ 1 problem (1 error, 0 warnings)
```

(by the way, as far as I know regexes, `*.svelte` is already included by `**/*.svelte`; double glob at the beginning acts as "with any depth from 0 to max")

**Comment by baseballyama** [comment]
Please provide a reproducible repository. I can’t tell where the problem lies since I don’t even know what version of the libraries you are using. By the way, since it works in my repository, I believe it’s a configuration issue.

**Comment by WarningImHack3r** [comment]
The first configuration of my last reply does actually work, isn’t it good enough?
I use the current latest version of everything, and the beta of `eslint-plugin-svelte`

**Comment by baseballyama** [comment]
I see. That’s great.

I’ll close this issue. When `eslint-plugin-svelte` v3 is released, I will update `sv` command's repo. Also, the README will be made easier to understand, so this kind of confusion should happen less often.

**Comment by WarningImHack3r** [comment]
Thanks for you hard work!!
